firebase-vertexai/src/main/kotlin/com/google/firebase/vertexai/type/ContentModality.kt

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.firebase.vertexai.type
+
+import com.google.firebase.vertexai.common.util.FirstOrdinalSerializer
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+public class ContentModality private constructor(public val ordinal: Int) {
+
+  @Serializable(Internal.Serializer::class)
+  internal enum class Internal {
+    @SerialName("MODALITY_UNSPECIFIED") UNSPECIFIED,
+    TEXT,
+    IMAGE,
+    VIDEO,
+    AUDIO,
+    DOCUMENT;
+
+    internal object Serializer : KSerializer<Internal> by FirstOrdinalSerializer(Internal::class)
+
+    internal fun toPublic() =
+      when (this) {
+        TEXT -> ContentModality.TEXT
+        IMAGE -> ContentModality.IMAGE
+        VIDEO -> ContentModality.VIDEO
+        AUDIO -> ContentModality.AUDIO
+        DOCUMENT -> ContentModality.DOCUMENT
+        else -> ContentModality.UNSPECIFIED
+      }
+  }
+
+  public companion object {
+    @JvmField public val UNSPECIFIED: ContentModality = ContentModality(0)
+
+    /** Plain text. */
+    @JvmField public val TEXT: ContentModality = ContentModality(1)
+
+    /** Image. */
```

Comment:
best javadocs this side of the atlantic 💯

Reply:
We're missing javadocs for the `UNSPECIFIED` field though 😅

Reply:
Oh, absolutely, adding it!